--- a/src/xfpm-battery.c
+++ b/src/xfpm-battery.c
@@ -30,6 +30,10 @@
 #include <string.h>
 #endif
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
 #include <gtk/gtk.h>
 #include <upower.h>
 
@@ -342,6 +346,7 @@
     guint state;
     gdouble percentage;
     guint64 to_empty, to_full;
+
     g_object_get(device,
 		 "is-present", &present,
 		 "percentage", &percentage,
@@ -370,6 +375,17 @@
     }
 }
 
+static gboolean
+read_bat0(gpointer data) {
+    int fd, b;
+    char c;
+
+    if ((fd = open("/sys/class/power_supply/BAT0/status", O_RDONLY)) != -1) {
+       b = read(fd, &c, 1);
+       close(fd);
+    }
+}
+
 static void
 xfpm_battery_changed_cb (UpDevice *device,
 #if UP_CHECK_VERSION(0, 99, 0)
@@ -378,6 +394,8 @@
 			 XfpmBattery *battery)
 {
     xfpm_battery_refresh (battery, device);
+    //Hack to refresh the battery status
+    g_timeout_add (1500, (GSourceFunc) read_bat0, NULL);
 }
 
 static void xfpm_battery_get_property (GObject *object,
