--- a/src/xfpm-xfconf.c
+++ b/src/xfpm-xfconf.c
@@ -492,7 +492,7 @@
                                                         NULL, NULL,
                                                         XFPM_DO_SUSPEND,
                                                         XFPM_DO_HIBERNATE,
-                                                        XFPM_DO_HIBERNATE,
+                                                        XFPM_DO_SUSPEND,
                                                         G_PARAM_READWRITE));
 
     /**
--- a/src/xfpm-power.c
+++ b/src/xfpm-power.c
@@ -511,8 +511,6 @@
     g_signal_emit (G_OBJECT (power), signals [WAKING_UP], 0);
     /* Check/update any changes while we slept */
     xfpm_power_get_properties (power);
-    /* Restore the brightness level from before we suspended */
-    xfpm_brightness_set_level (brightness, brightness_level);
 
 #ifdef WITH_NETWORK_MANAGER
     if ( network_manager_sleep )
